index.rst

```diff
@@ -1,26 +1,363 @@
+``https://(address)/api/central-availability/(service-name)`` .
 
-  See https://developer.lsst.io/restructuredtext/style.html
-  for a guide to reStructuredText writing.
+This service would return the prescribed XML response for every RSP service,
+by default based only on information known to the Kubernetes controller and on
+a future repository of planned-downtime information.
 
-  Do not put the title, authors or other metadata in this document;
-  those are automatically added.
+Optionally, for specifically configured services, the central availability service
+would, when it believed the underlying service to be up, call through to that
+service's own, internal ``/availability`` endpoint for a more thorough assessment,
+following the recommendation in the VOSI standard.
+This would allow, for instance, the TAP service to check that the underlying
+Qserv database was responding to simple requests, or the SODA service to check
+that the production Butler is functional.
 
-  Use the following syntax for sections:
+Kubernetes ingress rules would be used to redirect IVOA-style ``/availability``
+endpoints "below" the specific service address,
+such as https://(RSP-external-address)/api/tap/availability,
+away from the service-specific availability endpoint, and
+to the corresponding endpoint on the central availability service.
 
-  Sections
-  ========
+The central availability service could be directly exposed externally,
+or only through the redirected external per-service endpoints; this is TBD.
+An advantage to exposing it directly is that it provides a natural home for
+delivering status information for non-IVOA services (e.g., Nublado) that are
+not constrained to have their endpoints registered in VOSI-like ways.
+Any "internal" service-specific availability endpoints, provided by the specific
+IVOA service pods themselves, would not be exposed externally at all.
 
-  and
+For services for which throttling / rate-limiting are enabled, if that is
+implemented in a generic way across the RSP's web services, the central
+availability service should be able to be extended to return a message
+concerning whether rate-limiting is currently being applied.
 
-  Subsections
-  -----------
+Finally, with a uniform availability service in place, it is easy to imagine
+writing a lightweight "dashboard" / "Christmas tree" status display for all
+the components of the RSP.
 
-  and
+The proposal includes the following specific elements:
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+Basic /availability XML retrieval from Kubernetes controller
+------------------------------------------------------------
+
+The core element of the proposal is a service which can be configured with
+a list of services to represent, and the logic necessary to query the
+Kubernetes controller for the present state of a service and translate that
+into the ``available`` and ``upSince`` attributes of the VOSI-Availability
+data model.
```

Comment:
From a "what's the absolute minimum" perspective, that is surely true, but I view Rubin/LSST as a chance to try to deliver a slightly higher level of service where it seems useful, and if it's not costly in code.  I'll reword the sentence to avoid any implication that the `upSince`-handling is itself a "core" capability.  I was using "core" to mean that the "talk-to-Kubernetes" component is the core of the proposal.

Reply:
Changed accordingly.